devel/sdl12-compat/Portfile

```diff
@@ -26,13 +26,15 @@ depends_lib     port:libsdl2
 
 patchfiles      CMakeLists.txt.patch \
                 SDL2path.patch \
-                SDL_stdinc.h.patch
+                SDL_stdinc.h.patch \
+                patch-SDL_endian_h.diff
+
 post-patch {
     reinplace "s|__PREFIX__|${prefix}|" ${worksrcpath}/src/SDL12_compat.c
 }
 
 cmake.out_of_source yes
-configure.args-append   -DSDL2_PATH=${prefix}
+configure.args-append   -DSDL2_PATH=${prefix} -D__OpenBSD__=
```

Comment:
Defining a reserved identifier that indicates that we're on a different OS than we actually are is going to cause nothing but trouble. What problem are you trying to solve?

Reply:
> Defining a reserved identifier that indicates that we're on a different OS than we actually are is going to cause nothing but trouble. What problem are you trying to solve?

This problem:

https://github.com/dlbeer/quirc/blob/9c0f555acb2531d818f2c405044418fc75379fb2/demo/camera.c#L26-L30

Don't define the variable, get a Linux include. Define the variable and get something that compiles. It's the only appearance of this variable in upstream.

Reply:
Whoops, this problem too:

https://github.com/libsdl-org/sdl12-compat/blob/6d68bff0e950be9c550f3b3bc60e6fa4a86edd7d/include/SDL/SDL_endian.h#L41-L45

macOS uses `<machine/endian.h>`, which isn't represented upstream. Again, this is the only place this variable is used in compilation.

If it's not defined, then no `endian.h` can be found, and build like this one that use `libsdl_gfx` don't compile.

Whatever we do with the `quirc` port, some version of the fix this commit addresses should be merged.

Reply:
I used the `__APPLE__` variable, used elsewhere in upstream.